Godeps/_workspace/src/k8s.io/kubernetes/pkg/apis/extensions/v1beta1/conversion.go

```diff
@@ -84,6 +97,122 @@ func addConversionFuncs(scheme *runtime.Scheme) {
 	}
 }
 
+func Convert_v1beta1_ReplicaSet_to_api_ReplicationController(in *ReplicaSet, out *api.ReplicationController, s conversion.Scope) error {
+	if defaulting, found := s.DefaultingInterface(reflect.TypeOf(*in)); found {
+		defaulting.(func(*ReplicaSet))(in)
+	}
+	if err := api.Convert_unversioned_TypeMeta_To_unversioned_TypeMeta(&in.TypeMeta, &out.TypeMeta, s); err != nil {
+		return err
+	}
+	if err := v1.Convert_v1_ObjectMeta_To_api_ObjectMeta(&in.ObjectMeta, &out.ObjectMeta, s); err != nil {
+		return err
+	}
+	if err := Convert_v1beta1_ReplicaSetSpec_to_api_ReplicationControllerSpec(&in.Spec, &out.Spec, s); err != nil {
+		// TODO: Any better way to do this?
+		if !strings.Contains(err.Error(), "invalid label selector") {
+			return err
+		}
+		if out.Annotations == nil {
+			out.Annotations = make(map[string]string)
+		}
+		out.Annotations[nonConvertibleAnnotation] = ".spec.selector"
```

Comment:
I think the "convert anyway to allow no-touch round tripping which is detectable server-side" argument is strong enough to win.  I convinced myself while thinking through things like labelling.  I should still be allowed to issue updates to RS's that I don't understand through an RC endpoint to be able to handle non-conflicting paths.